resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -67,12 +68,19 @@ private[spark] class BasicExecutorFeatureStep(
     }
   private val executorLimitCores = kubernetesConf.get(KUBERNETES_EXECUTOR_LIMIT_CORES)
 
-  override def configurePod(pod: SparkPod): SparkPod = {
-    val name = s"$executorPodNamePrefix-exec-${kubernetesConf.roleSpecificConf.executorId}"
+  // If the driver pod is killed, the new driver pod will try to
+  // create new executors with the same name, but it will fail
+  // and hangs indefinitely because a terminating executors blocks
+  // the creation of the new ones, so to avoid that apply salt
+  private val executorNameSalt = Random.alphanumeric.take(4).mkString("").toLowerCase
```

Comment:
I think it is only generated when a new Spark Application is submitted. We need a random number which is regenerated for every new driver Pod not for new Application. Also it is too long because Kubernetes only allows pods with name length not longer than 64 character.

Reply:
> I think it is only generated when a new Spark Application is submitted. We need a random number which is regenerated for every new driver Pod not for new Application. Also it is too long because Kubernetes only allows pods with name length not longer than 64 character.

The application id is generated when then JVM launches - see `SchedulerBackend.scala`. Note this application ID isn't populated by spark submit itself.

Reply:
If we use applicationID as a salt the executor pod name will exceed the 64 length limit in case the application name is longer.
`override def configureExecutorPod(pod: SparkExecutorPod): SparkExecutorPod = {
    val name = s"$executorPodNamePrefix-$applicationID" +
        s"-exec-${kubernetesConf.roleSpecificConf.executorId}"`
For example if the application name is networkwordcount then this result an executor pod name like: `networkwordcount-1519234651100-spark-application-1528815922371-exec-10`

Reply:
We should just have the executor pod name be `s"$applicationId-exec-$executorId` then. Don't think the pod name prefix has to strictly be tied to the application name. The application name should be applied to a label so the executor pod can be located using that when using the dashboard and kubectl, etc.

Reply:
4 digits does not have enough randomness to be a reliable salt, which is why I suggest this to avoid collisions.